pkg/util/os/os.go

```diff
@@ -43,3 +47,36 @@ func CurrentOS() OS {
 	}
 	panic("Unexpected OS type")
 }
+
+func CurrentExecutable() (string, error) {
+	currentExec, err := osext.Executable()
+	if err != nil {
+		return "", err
+	}
+	return currentExec, nil
+}
+
+// EnvironmentAsMap returns os.Environ as map.
+func EnvironmentAsMap() map[string]string {
+	env := os.Environ()
+	envMap := make(map[string]string, len(env))
+
+	for _, setting := range env {
+		pair := strings.Split(setting, "=")
+		envMap[pair[0]] = pair[1]
+	}
+
+	return envMap
+}
+
+// EnvironmentToSlice converts a map containing environment settings
+// into a slice of strings where each element has the format <key>=<value>
+func EnvironmentToSlice(envMap map[string]string) []string {
```

Comment:
It is a bit a mirror to _EnvironmentAsMap_, but here I see more the point of creating a generic helper method, even though the '=' as separator makes it env specific again. The intend is really to make it a bit easier to work with the full process environment which is only returned as slice.